### ISSUE TITLE
💅 useOptionalChain unsafe fix incorrectly discards conditions in logical and chains

### ISSUE THREAD

**Issue by bryanrsmith** [biomejs/biome#1656]
### Environment information

```bash
CLI:
  Version:                      1.5.3
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.5.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/9.8.0"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### Rule name

useOptionalChain

### Playground link

https://biomejs.dev/playground/?code=dABhAGcAcwAgACYAJgAgAHQAYQBnAHMALgBpAG4AYwBsAHUAZABlAHMAKAAnAGEAJwApACAAJgAmACAAdABhAGcAcwAuAGkAbgBjAGwAdQBkAGUAcwAoACcAYgAnACkA

### Expected result

The unsafe fix currently applies:

```diff
- tags·&&·tags.includes('a')·&&·tags.includes('b')
+ tags?.includes('b')
```

The expected result is:

```diff
- tags·&&·tags.includes('a')·&&·tags.includes('b')
+ tags?.includes('a') && tags.includes('b')
```

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
